truss/tests/test_model_inference.py

```diff
@@ -232,6 +232,27 @@ def test_async_streaming():
         assert predict_non_stream_response.json() == "01234"
 
 
+@pytest.mark.integration
+def test_async_streaming_timeout():
+    with ensure_kill_all():
+        truss_root = Path(__file__).parent.parent.parent.resolve() / "truss"
+
+        truss_dir = truss_root / "test_data" / "test_streaming_read_timeout"
+
+        tr = TrussHandle(truss_dir)
+
+        _ = tr.docker_run(local_port=8090, detach=True, wait_for_server_ready=True)
+        truss_server_addr = "http://localhost:8090"
+        predict_url = f"{truss_server_addr}/v1/models/model:predict"
+
```

Comment:
would also be good to assert that this shows up as a timeout error in the logs (there are other reasons a stream might get terminated) -- could you use one of our helpers to do this? See https://github.com/basetenlabs/truss/blob/main/truss/tests/test_model_inference.py#L384 for an example